### ISSUE TITLE
Labels get renumbered 

### ISSUE THREAD

**Issue by nicochidt** [fastai/fastai#2264]
I found an issue when trying to create an SSD. I got an error when using a custom loss function produced by an automatic relabeling of the data. In [this gist](https://gist.github.com/nicochidt/0f5fb5344fa915e3961b53c6009ba0e4), I'm providing a minimal code example with the stacktrace showing the error. 

In particular, images labeled as (0,1) are transformed into (1,2) making the calculation of cross_entropy to fail. Prior to that, the images are correctly labeled (as can be checked using show_batch). 

In [this notebook](https://github.com/nicochidt/ml_examples/blob/master/fastai/bounding_boxes/Data%20relabeling.ipynb), there is more detailed information, showing the right labeling of images and how they are changed in the loss function. 

I'm not sure what is generating this. 

**Installation details:** 

```text
=== Software === 
python       : 3.6.4
fastai       : 1.0.55
fastprogress : 0.1.21
torch        : 1.1.0
torch cuda   : None / is **Not available** 

=== Hardware === 
No GPUs available 

=== Environment === 
platform     : Darwin-18.6.0-x86_64-i386-64bit
conda env    : Unknown
python       : /usr/local/opt/python/bin/python3.6
sys.path     : 
/usr/local/Cellar/python/3.6.4/Frameworks/Python.framework/Versions/3.6/lib/python36.zip
/usr/local/Cellar/python/3.6.4/Frameworks/Python.framework/Versions/3.6/lib/python3.6
/usr/local/Cellar/python/3.6.4/Frameworks/Python.framework/Versions/3.6/lib/python3.6/lib-dynload
/Users/nico/fastai/lib/python3.6/site-packages
no supported gpus found on this system
```

**Comment by sgugger** [comment]
The 0 is hardcoded to be background in fastai, and any label index gets added 1 yes. That's not a bug, that's to be used with the Focal loss implemented in the object detection notebook.

**Comment by nicochidt** [comment]
Then, how should be avoided the error in the cross_entropy calculation, @sgugger? I can add one extra output layer, but that's not what the output of the model should be. Why that error is not found in the lesson that I've pointed?

**Comment by sgugger** [comment]
The `ObjectItemList` from the fastai library is meant to be used with a model like a RetinaNet or a FPN that produce as many outputs as the final anchors, a lot of them being background. Your use case is different so you should use a different class (probably subclassing `ObjectItemList`).

I'm not sure which lesson notebook you have pointed to as there are no lessons in fastai v1 for object detection yet.
